content/courses/advanced-ada/chapters/subprograms.rst

```diff
@@ -6,21 +6,456 @@ Subprograms
 Expression functions
 --------------------
 
-.. todo::
+Usually, we implement Ada functions with a construct like this:
+:ada:`begin return X; end;`. In other words, we create a :ada:`begin ... end;`
+block and we have at least one :ada:`return` statement in that block. An
+expression function, in contrast, is a function that is implemented with a
+simple expression in parentheses, such as :ada:`(X);`. In this case, we don't
+use a :ada:`begin ... end;` block or a :ada:`return` statement.
 
-    Complete section!
+As an example of an expression, let's say we want to implement a function
+named :ada:`Is_Zero` that checks if the value of the integer parameter :ada:`I`
+is zero. We can implement this function with the expression :ada:`I = 0`. In
+the standard approach, we would create the implementation by writing
```

Comment:
"standard" suggests more than it should. Maybe use "usual" again, to tie it to the previous paragraph?